src/python/pants/engine/isolated_process.py

```diff
@@ -6,36 +6,44 @@
                         unicode_literals, with_statement)
 
 import logging
+import six
 
 from pants.engine.fs import EMPTY_SNAPSHOT, DirectoryDigest
 from pants.engine.rules import RootRule
-from pants.util.objects import TypeCheckError, datatype
+from pants.util.objects import Exactly, SubclassesOf, TypeCheckError, datatype
 
 
 logger = logging.getLogger(__name__)
 
+_default_timeout_seconds = 15 * 60
+
 
 class ExecuteProcessRequest(datatype([
   ('argv', tuple),
   ('env', tuple),
   ('input_files', DirectoryDigest),
-  ('output_files', tuple)
+  ('output_files', tuple),
+  # NB: timeout_seconds covers the whole remote operation including queuing and setup.
+  ('timeout_seconds', Exactly(float, int)),
+  ('description', SubclassesOf(*six.string_types)),
```

Comment:
Nit: Encouraging folks to set names for processes, and discouraging them from setting specific timeouts seems like a good idea. So swapping the order of these arguments would be good.

Also, `argv[0]` would probably make a better default `description` than "process".